meteor/server/api/cleanup.ts

```diff
@@ -0,0 +1,417 @@
+import { Evaluations } from '../../lib/collections/Evaluations'
+import { ExpectedMediaItems } from '../../lib/collections/ExpectedMediaItems'
+import { ExpectedPlayoutItems } from '../../lib/collections/ExpectedPlayoutItems'
+import { ExternalMessageQueue } from '../../lib/collections/ExternalMessageQueue'
+import { IngestDataCache } from '../../lib/collections/IngestDataCache'
+import { MediaObjects } from '../../lib/collections/MediaObjects'
+import { MediaWorkFlows } from '../../lib/collections/MediaWorkFlows'
+import { MediaWorkFlowSteps } from '../../lib/collections/MediaWorkFlowSteps'
+import { Organizations, OrganizationId } from '../../lib/collections/Organization'
+import { PartInstances } from '../../lib/collections/PartInstances'
+import { Parts } from '../../lib/collections/Parts'
+import { PeripheralDeviceCommands } from '../../lib/collections/PeripheralDeviceCommands'
+import { PeripheralDevices, PeripheralDeviceId } from '../../lib/collections/PeripheralDevices'
+import { Pieces } from '../../lib/collections/Pieces'
+import { RundownBaselineAdLibActions } from '../../lib/collections/RundownBaselineAdLibActions'
+import { RundownBaselineAdLibPieces } from '../../lib/collections/RundownBaselineAdLibPieces'
+import { RundownBaselineObjs } from '../../lib/collections/RundownBaselineObjs'
+import { RundownLayouts } from '../../lib/collections/RundownLayouts'
+import { RundownPlaylists, RundownPlaylistId, RundownPlaylist } from '../../lib/collections/RundownPlaylists'
+import { Rundowns, RundownId } from '../../lib/collections/Rundowns'
+import { Segments } from '../../lib/collections/Segments'
+import { ShowStyleBases } from '../../lib/collections/ShowStyleBases'
+import { ShowStyleVariants } from '../../lib/collections/ShowStyleVariants'
+import { Snapshots } from '../../lib/collections/Snapshots'
+import { Studios, StudioId } from '../../lib/collections/Studios'
+import { Timeline } from '../../lib/collections/Timeline'
+import { UserActionsLog } from '../../lib/collections/UserActionsLog'
+import { PieceInstances } from '../../lib/collections/PieceInstances'
+import { isAnyQueuedWorkRunning } from '../codeControl'
+import { getActiveRundownPlaylistsInStudioFromDb } from './studio/lib'
+import { ExpectedPackages } from '../../lib/collections/ExpectedPackages'
+import { ExpectedPackageWorkStatuses } from '../../lib/collections/ExpectedPackageWorkStatuses'
+import { PackageContainerPackageStatuses } from '../../lib/collections/PackageContainerPackageStatus'
+import { PackageInfos } from '../../lib/collections/PackageInfos'
+
+export function cleanupOldDataInner(actuallyCleanup: boolean = false): CollectionCleanupResult | string {
+	if (actuallyCleanup) {
+		const notAllowedReason = isAllowedToRunCleanup()
+		if (notAllowedReason) return `Could not run the cleanup function due to: ${notAllowedReason}`
+	}
+
+	/** Clean up stuff that are older than this: */
+	const MAXIMUM_AGE = 1000 * 60 * 60 * 24 * 100 // 100 days
```

Comment:
Feels like this should be configurable through `Meteor.settings` or CoreSystem object.

Reply:
Very good idea! 👍